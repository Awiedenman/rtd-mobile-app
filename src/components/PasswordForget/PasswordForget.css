.password-forget-link {
  color: #222222;
  margin: 20px;
}