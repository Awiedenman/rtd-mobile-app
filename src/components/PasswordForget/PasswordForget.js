import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import { auth } from '../../firebase';
import * as routes from '../../constants/routes';

const PasswordForgetPage = () =>
  <div> 
    <h1>PasswordForget</h1>
    <PasswordForgetForm />
  </div>;

const INITIAL_STATE = {
  email: '',
  error: null
};

class PasswordForgetForm extends Component {
  constructor(props) {
    super(props);
    this.state = { ...INITIAL_STATE };
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  }

  onSubmit = (event) => {
    const { email } = this.state;

    auth.doPasswordReset(email)
      .then(() => {
        this.setState({ ...INITIAL_STATE });
      })
      .catch(error => {
        this.setState({error: error});
      });

    event.preventDefault();
  }

  render() {
    const {
      email,
      error
    } = this.state;

    const isInvalid = email === '';

    return (
      <form onSubmit={this.onSubmit}>
        <input
          className='password-forget'
          name="email"
          value={email}
          onChange={this.handleChange}
          type="email"
          placeholder="Email Address"
        />
        <button disabled={isInvalid} type="submit">
          Reset My Password
        </button>

        { error && <p>{error.message}</p> }
      </form>
    );
  }
}

export const PasswordForgetLink = () =>
  <p>
    <Link to={routes.PASSWORD_FORGET}>Forgot Password?</Link>
  </p>;

export default PasswordForgetPage;

export {
  PasswordForgetForm
  // PasswordForgetLink
};