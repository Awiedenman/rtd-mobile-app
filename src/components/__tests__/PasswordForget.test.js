import React from 'react';
import { PasswordForgetPage, PasswordForgetLink } from '../PasswordForget/PasswordForget';
// import { PasswordForgetForm, PasswordForgetLink } from '../PasswordForget/PasswordForget';
import { shallow } from 'enzyme';

describe('PasswordForget', () => {
  it('should matchSnapshot', () => {
    const wrapper = shallow(<PasswordForgetPage/>);

    expect(wrapper).toMatchSnapshot();
  });

  describe('handleChange', () => {
    it('should update handleChange', () => {
      const wrapper = shallow( < PasswordForgetPage /> );
      const handleChange = jest.fn();
      const event = {
        target: {
          name: 'email',
          value: 'austin@gmail.com'
        }
      };

      handleChange(event);

      wrapper.find('form').simulate('change', event);
      wrapper.find('form').simulate('change', event);
      expect(handleChange).toHaveBeenCalled();
      // expect(wrapper.state('name')).toEqual('email');
      expect(wrapper.state('email')).toEqual('austin@gmail.com');

    });
  });

  describe('onSubmit', () => {
    it('should call onSubmit on submission of the input field ', () => {
      const wrapper = shallow(<PasswordForgetPage/>);
      const onSubmit = jest.fn();
      const event = {
        preventDefault() {},
        target: {
          value: ''
        }
      };
      
      onSubmit(event);
      wrapper.find('form').simulate('sumbit', event);
      expect(onSubmit).toHaveBeenCalled();
    });
  });

  describe('passwordForgetLink', () => {
    it('should render a link to the forgot password component ', () => {
      const wrapper = shallow(<PasswordForgetLink />);
  
      expect(wrapper).toMatchSnapshot();
    });
    
  });
  
});



