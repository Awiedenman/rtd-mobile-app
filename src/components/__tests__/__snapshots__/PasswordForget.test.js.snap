// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordForget passwordForgetLink should render a link to the forgot password component  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        replace={false}
        to="/pw-forget"
>
        Forgot Password?
</Link>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Forgot Password?",
        "replace": false,
        "to": "/pw-forget",
      },
      "ref": null,
      "rendered": "Forgot Password?",
      "type": [Function],
    },
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          to="/pw-forget"
>
          Forgot Password?
</Link>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Forgot Password?",
          "replace": false,
          "to": "/pw-forget",
        },
        "ref": null,
        "rendered": "Forgot Password?",
        "type": [Function],
      },
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`PasswordForget should matchSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PasswordForgetPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="password-forget"
          name="email"
          onChange={[Function]}
          placeholder="Email Address"
          type="email"
          value=""
/>,
        <button
          disabled={true}
          type="submit"
>
          Reset My Password
</button>,
        null,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "password-forget",
          "name": "email",
          "onChange": [Function],
          "placeholder": "Email Address",
          "type": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset My Password",
          "disabled": true,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Reset My Password",
        "type": "button",
      },
      null,
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="password-forget"
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            type="email"
            value=""
/>,
          <button
            disabled={true}
            type="submit"
>
            Reset My Password
</button>,
          null,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "password-forget",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email Address",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset My Password",
            "disabled": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Reset My Password",
          "type": "button",
        },
        null,
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
